examples/nextjs/src/app/[[...slug]]/page.js

```diff
@@ -10,28 +11,41 @@ const client = dotcmsClient.init({
     }
 });
 
-import { MyPage } from '@/components/my-page';
-
 export async function generateMetadata({ params, searchParams }) {
-    const data = await client.page.get({
+
+    const requestData = {
         path: params?.slug ? params.slug.join('/') : 'index',
         language_id: searchParams.language_id,
         'com.dotmarketing.persona.id': searchParams['com.dotmarketing.persona.id'] || '',
-        mode: searchParams.mode
-    });
+        mode: searchParams.mode,
+    };
+
+    // Add variantName to requestData if it exists and is not 'DEFAULT'
+    if (searchParams['variantName'] && searchParams['variantName'] !== 'DEFAULT') {
+        requestData.variantName = searchParams['variantName'];
+    }
+
+    const data = await client.page.get(requestData);
 
     return {
         title: data.entity.page.friendlyName || data.entity.page.title
     };
 }
 
 export default async function Home({ searchParams, params }) {
-    const data = await client.page.get({
+    const requestData = {
         path: params?.slug ? params.slug.join('/') : 'index',
         language_id: searchParams.language_id,
-        personaId: searchParams['com.dotmarketing.persona.id'] || '',
-        mode: searchParams.mode
-    });
+        'com.dotmarketing.persona.id': searchParams['com.dotmarketing.persona.id'] || '',
+        mode: searchParams.mode,
+    };
+
+    // Add variantName to requestData if it exists and is not 'DEFAULT'
+    if (searchParams['variantName'] && searchParams['variantName'] !== 'DEFAULT') {
+        requestData.variantName = searchParams['variantName'];
+    }
```

Comment:
Move the fallback to the library so we don't have to ask the developer to do that.

Reply:
to `client.page.get`?

Reply:
Handled inside @dotcms/client